pkgs/misc/vim-plugins/update.py

```diff
@@ -386,6 +416,31 @@ def generate_nix(plugins: List[Tuple[str, str, Plugin]], outfile: str):
     print(f"updated {outfile}")
 
 
+def update_redirects(input_file: Path, output_file: Path, redirects: dict):
+    with fileinput.input(input_file, inplace=True) as f:
+        for line in f:
+            print(redirects.get(line, line), end="")
+    print(
+        f"""\
+Redirects have been detected and {input_file} has been updated. Please take the
+following steps:
+    1. Go ahead and commit just the updated expressions as you intended to do:
+            git add {output_file}
+            git commit -m "vimPlugins: Update"
+    2. If any of the plugin names were changed, add the old names as aliases in
+    aliases.nix
+    3. Make sure the updated {input_file} is still correctly sorted:
+            sort -udf ./vim-plugin-names > sorted && mv sorted vim-plugin-names
+    4. Run this script again so these changes will be reflected in the
+    generated expressions (no need to use the --update-redirects flag again):
+            ./update.py
```

Comment:
It would be nicer to just generate the nice expressions in the first place. I suspect this step would be easy to miss and even easier to miss in review.

Reply:
It would be nice if multiple invocations were not necessary but I think this script would then need to automatically commit the update before updating the redirects. It seems like if this was all done at once in one commit it would be harder hard to review than if the updates were separated from the redirect updates as I suggest.

Reply:
Yes, committing both changes at the same time is not ideal. But I think it is better than accidentally having the two files go out of sync, which I think is likely with the current model.